From 0fcd6c67a5fa9bcad4f9c0c9371dbd72dd8e0512 Mon Sep 17 00:00:00 2001
From: m <m@l>
Date: Sat, 15 Aug 2026 19:57:26 -0400
Subject: [PATCH] 
 5-models-server/0016-laguna-deduplicate-definitions-vs-the-early-snapshot.patch

---
 src/llama-model.cpp |  1 -
 src/llama-model.h   |  1 -
 src/models/models.h | 12 ------------
 3 files changed, 14 deletions(-)

diff --git a/src/llama-model.cpp b/src/llama-model.cpp
index 20598b36e..a29e21d46 100644
--- a/src/llama-model.cpp
+++ b/src/llama-model.cpp
@@ -848,7 +848,6 @@ const char * llm_type_name(llm_type type) {
         case LLM_TYPE_100B_A6B:      return "100B.A6B";
         case LLM_TYPE_102B_A12B:     return "102B.A12B";
         case LLM_TYPE_106B_A12B:     return "106B.A12B";
-        case LLM_TYPE_118B_A8B:      return "118B.A8B";
         case LLM_TYPE_120B_A12B:     return "120B.A12B";
         case LLM_TYPE_122B_A10B:     return "122B.A10B";
         case LLM_TYPE_196B_A11B:     return "196B.A11B";
diff --git a/src/llama-model.h b/src/llama-model.h
index 4635ffd53..ab313499b 100644
--- a/src/llama-model.h
+++ b/src/llama-model.h
@@ -132,7 +132,6 @@ enum llm_type {
     LLM_TYPE_100B_A6B,
     LLM_TYPE_102B_A12B, // Solar-Open
     LLM_TYPE_106B_A12B, // GLM-4.5-Air
-    LLM_TYPE_118B_A8B,  // Laguna-S-2
     LLM_TYPE_120B_A12B, // Nemotron 3 Super
     LLM_TYPE_122B_A10B, // Qwen3.5
     LLM_TYPE_196B_A11B, // Step3.5-Flash
diff --git a/src/models/models.h b/src/models/models.h
index 0f0c761d8..1113d6105 100644
--- a/src/models/models.h
+++ b/src/models/models.h
@@ -1363,18 +1363,6 @@ struct llama_model_dflash : public llama_model_base {
 };
 
 
-struct llama_model_laguna : public llama_model_base {
-    llama_model_laguna(const struct llama_model_params & params) : llama_model_base(params) {}
-    void load_arch_hparams(llama_model_loader & ml) override;
-    void load_arch_tensors(llama_model_loader & ml) override;
-
-    struct graph : public llm_graph_context {
-        graph(const llama_model & model, const llm_graph_params & params);
-    };
-
-    std::unique_ptr<llm_graph_context> build_arch_graph(const llm_graph_params & params) const override;
-};
-
 
 struct llama_model_mistral4 : public llama_model_deepseek2 {
     llama_model_mistral4(const struct llama_model_params & params) : llama_model_deepseek2(params) {}
